### PR TITLE
Update QUICHE from c2576eff3 to 3743c9285

### PR DESCRIPTION
**Pull request by RenjieTang**
https://github.com/google/quiche/compare/c2576eff3..3743c9285

$ git log c2576eff3..3743c9285 --date=short --no-merges --format="%ad %al %s"

2022-09-13 renjietang Use NiceMock for MockRandom to fix test failures in Envoy.
2022-09-12 quiche-dev Internal change
2022-09-12 quiche-dev Internal change
2022-09-12 martinduke ConnectionIdGenerator that uses QUIC-LB or HASH_QUIC depending on config.
2022-09-12 birenroy Makes the `FLAGS_` prefix implicit in calls to `GetQuicFlag()`.
2022-09-09 wub Internal change
2022-09-09 martinduke Internal change
2022-09-09 vasilvv Migrate QuicGoogleClient to use QuicEventLoop.
2022-09-09 quiche-dev Collapse InformationalResponseInternal class into public InformationalResponse.
2022-09-09 wub Allow caller of QuicFramer::BuildIetfStatelessResetPacket to provide a random bytes generator.
2022-09-09 quiche-dev Internal change
2022-09-09 quiche-dev Pass QUIC client connection ID to ProofSource.
2022-09-08 ramosalex Add bind to interface socket option support to QuicClientDefaultNetworkHelper.
2022-09-08 renjietang Populate stats needed for multi-port.
2022-09-07 diannahu Drop buffered metadata for a stream in ForgetStream().
2022-09-06 martinduke Fix incorrect flag checks in quic_connection_test.
2022-09-06 bnc Move QuicIpAddress and IpAddressFamily to third_party/quiche.
2022-09-06 martinduke Fix broken tests.
2022-09-06 quiche-dev Make binary_http library available to external bazel users.
2022-09-06 bnc Enabling rolled out flags.
2022-09-06 martinduke Extend LoadBalancerEncoder to support ConnectionIdGeneratorInterface.
2022-09-06 bnc Add missing includes.
2022-09-06 martinduke Automated g4 rollback of changelist 471573005.
2022-09-06 vasilvv Remove the use of QuicEpoll in qbone_session_test
2022-09-02 quiche-dev Fix file descriptor leak (on `bind()` failure) in QUIC client.
2022-09-02 ericorth Simplify higher-level socket interfaces
2022-09-02 quiche-dev Convert quic_bounded_crypto_send_buffer to a QUIC_PROTOCOL_FLAG
2022-09-01 quiche-dev Fix broken bazel build.
2022-09-01 quiche-dev Add absl/cleanup to dependencies for quiche_core.
2022-09-01 fayang Deprecate gfe2_reloadable_flag_quic_fix_bytes_accounting_for_buffered_coalesced_packets.
2022-09-01 martinduke Change LoadBalancerEncoder::Create to return unique_ptr.
2022-09-01 martinduke Update QuicConnection to Use ConnectionIdGeneratorInterface.
2022-09-01 quiche-dev Stop calling memcpy when `bytes_to_copy` is 0
2022-08-31 vasilvv Split quic_client_lib into quic_client_lib and quic_epoll_client_lib
2022-08-31 wub Internal change
2022-08-30 quiche-dev Open source Binary Http.
2022-08-30 vasilvv Fix and re-enable tests disabled in cl/468791547

Risk Level: Low
Testing: Existing tests
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

File: source/common/quic/envoy_deterministic_connection_id_generator.cc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+absl::optional<quic::QuicConnectionId>
+EnvoyDeterministicConnectionIdGenerator::GenerateNextConnectionId(
+    const quic::QuicConnectionId& original) {
+  auto new_cid = DeterministicConnectionIdGenerator::GenerateNextConnectionId(original);
+  if (!new_cid.has_value()) {
+    return new_cid;
+  }
+  adjustNewConnectionIdForRoutine(new_cid.value(), original);
+  return new_cid;
```
Comment (RyanTheOptimist) [inline source/common/quic/envoy_deterministic_connection_id_generator.cc:16]: Totally you call, but we could save a line by doing:

```
  if (new_cid.has_value()) {
    adjustNewConnectionIdForRoutine(new_cid.value(), original);
  }
  return new_cid;
```

(And below)

**Review by RyanTheOptimist — APPROVED** [review]
LGTM. This looks like a heck of a saga. Thanks for doing this!

File: source/common/quic/envoy_deterministic_connection_id_generator.cc (hunk 1 above)
Comment (RenjieTang) [reply source/common/quic/envoy_deterministic_connection_id_generator.cc:15]: done

**Review by RyanTheOptimist — DISMISSED** [review]

**Comment by moderation** [comment]
/lgtm deps

But looks like you have some CI errors

**Comment by alyssawilk** [comment]
Almost there!
I think compile time options build need  #ifdef ENVOY_ENABLE_QUIC around QUIC code to make sure Envoy and core tests still pass with QUIC built out.

**Review by RyanTheOptimist — DISMISSED** [review]

**Comment by alyssawilk** [comment]
Code coverage for source/common/quic is lower than limit of 91.8 (91.7) :-(

I'd say we can bump coverage down, and add a TODO for someone on our team to add unit tests and bump it back up.
I could take that on, as Renjie has effectively done my merge rotation as well as his own.

**Comment by alyssawilk** [comment]
https://github.com/envoyproxy/envoy/issues/23128 for tracking

**Comment by RyanTheOptimist** [comment]
@RenjieTang it looks like DCO is broken. Mind fixing and re-pushing?

**Review by RyanTheOptimist — APPROVED** [review]

**Pull request merged**